pkg/prowgen/prowgen_tests.go

```diff
@@ -201,22 +203,30 @@ func createTest(r Repository, line string, e2e E2ETest, tests *[]Test, commands
 			return fmt.Errorf("[%s] failed to match test %s: %w", r.RepositoryDirectory(), e2e.Match, err)
 		}
 		if matches && !commands.Has(line) {
-			*tests = append(*tests, Test{Command: line, OnDemand: e2e.OnDemand, IgnoreError: e2e.IgnoreError, RunIfChanged: e2e.RunIfChanged, SkipCron: e2e.SkipCron})
+			*tests = append(*tests, Test{Command: line, OnDemand: e2e.OnDemand, IgnoreError: e2e.IgnoreError, RunIfChanged: e2e.RunIfChanged, SkipCron: e2e.SkipCron, SkipImages: e2e.SkipImages})
 			commands.Insert(line)
 		}
 	}
 
 	return nil
 }
 
-func dependenciesFromImages(images []cioperatorapi.ProjectDirectoryImageBuildStepConfiguration) []cioperatorapi.StepDependency {
+func dependenciesFromImages(r Repository, images []cioperatorapi.ProjectDirectoryImageBuildStepConfiguration, testName string, skipImages []string) []cioperatorapi.StepDependency {
 	deps := make([]cioperatorapi.StepDependency, 0, len(images))
 	for _, image := range images {
-		dep := cioperatorapi.StepDependency{
-			Name: strings.ReplaceAll(string(image.To), "_", "-"),
-			Env:  strings.ToUpper(strings.ReplaceAll(string(image.To), "-", "_")),
+		imageFinal := strings.ReplaceAll(string(image.To), "_", "-")
+		if shouldAcceptImage(testName, skipImages, imageFinal) {
+			dep := cioperatorapi.StepDependency{
+				Name: imageFinal,
+				Env:  strings.ToUpper(strings.ReplaceAll(string(image.To), "-", "_")),
+			}
+			deps = append(deps, dep)
 		}
-		deps = append(deps, dep)
 	}
 	return deps
 }
+
+// Skip an image if it is in the skip image list
+func shouldAcceptImage(testName string, skipImages []string, image string) bool {
```

Comment:
maybe `shouldSkipImage` as the parameter is named that way? Then the caller can negate to achieve the same.

Reply:
LGTM. Aside from this little thing.

Reply:
Actually the comment was not ideally phrased. The method name is correct
since we accept it if `slices.Index(skipImages, image) < 0` which means the image is not in the skip images list.

Reply:
Yes I got that, I just meant it probably could be good to rename the method and invert the comparison, so that it is clear that this refers to the config `skipImages`.
But this way is also fine, up to you if you want to change or not :)